impeller/docs/standalone_gles.md

```diff
@@ -0,0 +1,195 @@
+
+static std::vector<std::shared_ptr<fml::Mapping>>
+ShaderLibraryMappingsForApplication() {
+  return {
+      std::make_shared<fml::NonOwnedMapping>(
+          impeller_fixtures_shaders_gles_data,
+          impeller_fixtures_shaders_gles_length),
+  };
+}
+
+// In the setup routine.
+
+auto mappings = ShaderLibraryMappingsForApplication();
+```
+
+And that's it. You have all the ingredients necessary to create a context. Create one now.
+
+```c++
+auto context = ContextGLES::Create(
+    std::move(gl),        // proc table
+    std::move(mappings),  // shader libraries
+    false                 // enable tracing
+);
+```
+
+Now for a tricky bit about OpenGL ES. Impeller is multi-threaded. Even when using OpenGL ES, objects above the Impeller HAL can be created, used, and consumed on any thread. But OpenGL ES isn't. Impeller also doesn't know anything about EGL. So it is your responsibility to tell Impeller which threads are safe to use OpenGL ES on.
+
+In our little toy setup, we only have a single thread. We tell Impeller which thread to use by creating a subclass of an `impeller::ReactorGLES::Worker`. Let's create a simple worker:
+
+```c++
+class ReactorWorker final : public impeller::ReactorGLES::Worker {
+ public:
+  ReactorWorker() = default;
+
+  // |ReactorGLES::Worker|
+  ~ReactorWorker() override = default;
+
+  ReactorWorker(const ReactorWorker&) = delete;
+
+  ReactorWorker& operator=(const ReactorWorker&) = delete;
+
+  // |ReactorGLES::Worker|
+  bool CanReactorReactOnCurrentThreadNow(
+      const ReactorGLES& reactor) const override {
+    return true;
+  }
+};
+```
+
+Add an instance of this reactor worker to the context. Whew, that was unnecessarily complicated. But this is only necessary for OpenGL ES because of its tricky threading. Skip this step for Vulkan and Metal. Those APIs are already thread safe.
+
+```c++
+context->AddReactorWorker(worker);
+```
+
+Once you have the context, use it to crate a renderer.
+
+```c++
+auto renderer = std::make_shared<Renderer>(context);
+```
+
+And setup is done. Keep the context and renderer around. We will be using it during frame rendering.
+
+# Rendering Frames
+
+Rendering frames is a matter of:
+* Wrapping the onscreen texture as a Surface (`impeller::SurfaceGLES` in our case).
+* Giving the surface to our renderer to render.
+* The renderer will prepare a render target from this surface and invoke a callback that we'll use to setup a render pass directed at that render target.
+* Populate the render pass how we see fit.
+* Tell our reactor worker that all operations need to be flushed to OpenGL.
+* Present the onscreen surface.
+* Repeat.
+
+## Wrap the Onscreen Surface
+
+Per frame, the onscreen surface can be wrapped using SurfaceGLES::WrapFBO where the default framebuffer in our case if FBO 0. Take care to to ensure that the pixel format matches the one we used to choose the EGL config. Figuring out the pixel size is left as an exercise for the reader.
```

Comment:
```suggestion
Per frame, the onscreen surface can be wrapped using SurfaceGLES::WrapFBO where the default framebuffer in our case is FBO 0. Take care to ensure that the pixel format matches the one we used to choose the EGL config. Figuring out the pixel size is left as an exercise for the reader.
```

Reply:
was this change applied? i still see the if/is error in the text i'm reviewing

Reply:
Sorry. The push I thought pushed failed due a confict. It should be fixed now.